File: php/features.php
<?php
declare(strict_types=1);

// Maplestory SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MaplestoryFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MaplestoryBaseFeature();
            case "test":
                return new MaplestoryTestFeature();
            default:
                return new MaplestoryBaseFeature();
        }
    }
}

File: py/maplestory_sdk/features.py
# Maplestory SDK feature factory

from maplestory_sdk.feature.base_feature import MaplestoryBaseFeature
from maplestory_sdk.feature.test_feature import MaplestoryTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MaplestoryBaseFeature(),
        "test": lambda: MaplestoryTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
